src/Lachain.Core/Blockchain/Operations/BlockManager.cs

```diff
@@ -150,7 +164,35 @@ public Block LatestBlock()
         [MethodImpl(MethodImplOptions.Synchronized)]
         public Block? GetByHeight(ulong blockHeight)
         {
-            return _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHeight(blockHeight);
+            try
+            {
+                /* try to block from cache */
+                if (_heightCache.TryGetValue(blockHeight, out var block))
+                    return block;
+
+                var newBlock = _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHeight(blockHeight);
+
+                /* if it reach cache size limit, remove the oldest one */
+                if (_heightCacheQueue.Count == _blockSizeLimit)
+                {
+                    /* remove from queue */
+                    var oldestKey = _heightCacheQueue.Dequeue();
```

Comment:
Do you think this will fail for : 
_blockSizeLimit = 0 

Should we allow _blockSizeLimit = 0 ? I think - yes - as this would mean that no caching is being done.

Reply:
Hi @HMAshiqulIslam 
_blockSizeLimit = 0 means no caching enabled.
But _blockSizeLimit is being set in config.json. And the default value is 100. So caching is enabled by default.

Reply:
I meant : The function will probably not work when _blockLimitSize = 0. It will always return null. Can you please test it ?

Reply:
Hi @HMAshiqulIslam I will check it

Reply:
Please check my latest commit @sgladkov @HMAshiqulIslam

Reply:
I think - when we set _blockSizeLimit = 0, the items in the cache will never be removed due to this equality 
check: **_heightCacheQueue.Count == _blockSizeLimit**
You can test it locally. I think - every blocks will be stored in the cache which will increase the memory usage gradually.

Reply:
```
if (_blockSizeLimit == 0)
{
    return _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHeight(blockHeight);
}

try
{
    /* try to get block from cache */
    if (_heightCache.TryGetValue(blockHeight, out var block))
        return block;

    var newBlock = _stateManager.LastApprovedSnapshot.Blocks.GetBlockByHeight(blockHeight);

    /* if it reach cache size limit, remove the oldest one */
    if (_heightCacheQueue.Count == _blockSizeLimit)
    ...
```
Looks good now?

Reply:
Looks perfect now. Thanks.